mayastor/src/subsys/nvmf/subsystem.rs

```diff
@@ -64,7 +64,21 @@ impl Display for SubType {
     }
 }
 
+/// A wrapper around the SPDK NVMF subsystem.
+///
+/// # Safety
+///
+/// SPDK considers this structure threadsafe. This means this structure can be Send/Sync.
+///
+/// See more: https://github.com/spdk/spdk/blob/312a9d603dfc7dfb7061f33fa3f1fec766579dc7/lib/nvmf/nvmf_internal.h#L264-L305
 pub struct NvmfSubsystem(pub(crate) NonNull<spdk_nvmf_subsystem>);
+
+// Safety: SPDK considers the inner `spdk_nvmf_subsystem` threadsafe.
+unsafe impl Send for NvmfSubsystem {}
+
+// Safety: SPDK considers the inner `spdk_nvmf_subsystem` threadsafe.
+unsafe impl Sync for NvmfSubsystem {}
```

Comment:
Again, don't think the subsystem is sync. Two threads can change the subsystem at any given time.

Reply:
I see at least partial consideration being given to threading in things like state changes. 

https://github.com/openebs/spdk/blob/312a9d603dfc7dfb7061f33fa3f1fec766579dc7/lib/nvmf/subsystem.c#L448  

https://github.com/openebs/spdk/blob/312a9d603dfc7dfb7061f33fa3f1fec766579dc7/lib/nvmf/subsystem.c#L728

So it does seem like this type is sync?